src/new-components/banner/banner.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react'
+import { Box } from '../box'
+import { Columns, Column } from '../columns'
+
+import styles from './banner.module.css'
+
+export type BannerTone = 'info' | 'promotion'
+
+type BannerProps = {
+    id?: string
+    /**
+     * The tone of the Banner. Affects the background color and the outline.
+     */
+    tone: BannerTone
+    /**
+     * The icon that should be added inside the Banner.
+     */
+    icon: React.ReactChild
+    /**
+     * The title to be displayed at the top of the Banner.
+     */
+    title: string
+    /**
+     * An optional description to be displayed inside the Banner.
+     */
+    description?: string
+    /**
+     * An optional action to displayed inside the Banner.
+     */
+    action?: React.ReactChild
+}
+
+function Banner({ id, tone, icon, title, description, action }: BannerProps) {
```

Comment:
Anticipating that we might need it, I'd recommend that you enable forward-refs to this component's container:

```tsx
const Banner = forwardRef<HTMLDivElement, BannerProps>(
    function Banner({ … }, ref) {
        // pass ref={ref} to the outer `Box`
    }
)
```